graphbolt/src/csc_sampling_graph.cc

```diff
@@ -781,10 +783,15 @@ inline int64_t LaborPick(
     std::iota(picked_data_ptr, picked_data_ptr + num_neighbors, offset);
     return num_neighbors;
   }
-  torch::Tensor heap_tensor = torch::empty({fanout * 2}, torch::kInt32);
   // Assuming max_degree of a vertex is <= 4 billion.
-  auto heap_data = reinterpret_cast<std::pair<float, uint32_t>*>(
-      heap_tensor.data_ptr<int32_t>());
+  std::array<std::pair<float, uint32_t>, StackSize> heap;
+  auto heap_data = heap.data();
+  torch::Tensor heap_tensor;
+  if (fanout > StackSize) {
+    heap_tensor = torch::empty({fanout * 2}, torch::kInt32);
```

Comment:
Is this memory safe? `heap_tensor` holds the memory and it ends it's life outside of the `if` statement. Such memory will be deleted automatically even the data pointer is saved to `heap_data`?

Reply:
heap_tensor is defined outside of if.

Reply:
utilizing `torch:empty` instead of `new` or `std::vector` is to ask torch to help us manage memory better?

Reply:
Yeah, I thought that since we are relying on torch for everything, we can do the same here. Not sure if torch allocation is faster or new is faster.